topic-operator/src/main/java/io/strimzi/operator/topic/TopicCommandSerde.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.topic;
+
+import io.apicurio.registry.utils.kafka.SelfSerde;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.UncheckedIOException;
+
+/**
+ * TopicCommand Kafka Serde
+ */
+public class TopicCommandSerde extends SelfSerde<TopicCommand> {
+
+    @Override
+    public byte[] serialize(String topic, TopicCommand data) {
+        try {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
Since we're already representing the topic as JSON I think it would be better to represent the command as JSON too, then this would pretty much be a call to Jackson's `JsonMapper`.

I also wonder about including an explicit version.

Reply:
Explicit version?

Reply:
Yeah. I considered it at the time for the `TopicSerialization` and decided that no version was semantically equivalent to having a `version=0`. That's still true, of course, so I'm not saying you should do this. IOW, I'm still wondering. Only a need to change the format is really going to force me to decide ;-)